# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: ea966e78b296
# Full Hash: ea966e78b296be37881d4fa36f7cc805e7bbac82
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-07 03:53:09
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 6.3: Replace many nsILoadContext mutators with assertions, r=smaug
#   
#   Many of these setters are redundant, and duplicate work done elsewhere. As it is
#   no longer correct to set nsILoadContext attributes at these times, they're being
#   replaced with assertions.
# ==============================================================================

diff -r cb88e0bbb3b9 -r ea966e78b296 dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Mon Apr 06 14:29:50 2020 +0000
+++ b/dom/ipc/ContentParent.cpp	Mon Apr 06 14:29:52 2020 +0000
@@ -4792,14 +4792,8 @@
     newBrowserHost->GetBrowsingContext()->SetName(aName);
   }
 
-  // Don't send down the OriginAttributes if the content process is handling
-  // setting up the window for us. We only want to send them in the async case.
-  //
-  // If we send it down in the non-async case, then we might set the
-  // OriginAttributes after the document has already navigated.
-  if (!aSetOpener) {
-    Unused << newBrowserParent->SendSetOriginAttributes(openerOriginAttributes);
-  }
+  MOZ_ASSERT(newBrowserHost->GetBrowsingContext()->OriginAttributesRef() ==
+             openerOriginAttributes);
 
   if (aURIToLoad && aLoadURI) {
     nsCOMPtr<mozIDOMWindowProxy> openerWindow;